numpy/core/src/multiarray/array_method.c

```diff
@@ -819,39 +817,36 @@ unmasked_ufunc_loop_as_masked(
 
         /* Process unmasked values */
         mask = npy_memchr(mask, 0, mask_stride, N, &subloopsize, 0);
-        strided_loop(dataptrs, &subloopsize, strides, innerloopdata);
+        if (strided_loop(context,
+                dataptrs, &subloopsize, strides, strided_loop_auxdata) != 0) {
+            return -1;
```

Comment:
Maybe store the return value and compare it to 0, then return the value on error. This might make it future-proof